.github/workflows/OSX.yml

```diff
@@ -149,12 +149,25 @@ jobs:
           BUILD_CERTIFICATE_BASE64: ${{ secrets.OSX_CODESIGN_BUILD_CERTIFICATE_BASE64 }}
           P12_PASSWORD: ${{ secrets.OSX_CODESIGN_P12_PASSWORD }}
           KEYCHAIN_PASSWORD: ${{ secrets.OSX_CODESIGN_KEYCHAIN_PASSWORD }}
+          TEAM_ID : ${{ secrets.OSX_NOTARIZE_TEAM_ID }}
+          APPLE_ID: ${{ secrets.OSX_NOTARIZE_APPLE_ID }}
+          PASSWORD: ${{ secrets.OSX_NOTARIZE_PASSWORD }}
         run: |
           if [[ "$GITHUB_REPOSITORY" = "duckdb/duckdb" ]] ; then
             . scripts/osx_import_codesign_certificate.sh
-
-            codesign --all-architectures --force --sign "Developer ID Application: Stichting DuckDB Foundation" build/release/duckdb
-            codesign --all-architectures --force --sign "Developer ID Application: Stichting DuckDB Foundation" build/release/src/libduckdb*.dylib
+            
+            echo -e '<?xml version="1.0" encoding="UTF-8"?>\n<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">\n<plist version="1.0">\n<dict>\n  <key>com.apple.security.cs.disable-library-validation</key>\n  <true/>\n</dict>\n</plist>' > entitlements.plist
+            codesign --options runtime --entitlements entitlements.plist --all-architectures --force --sign "Developer ID Application: Stichting DuckDB Foundation" build/release/duckdb
+            codesign --options runtime --entitlements entitlements.plist --all-architectures --force --sign "Developer ID Application: Stichting DuckDB Foundation" build/release/src/libduckdb*.dylib
+          
+            zip -j notarize.zip build/release/duckdb build/release/src/libduckdb.dylib 
+            export XCRUN_RESPONSE=$(xcrun notarytool submit --progress --apple-id "$APPLE_ID" --password "$PASSWORD" --team-id "$TEAM_ID" --wait -f json notarize.zip)
+            rm notarize.zip
+            if [[ $(echo $XCRUN_RESPONSE | ./build/release/duckdb -csv -noheader -s "FROM read_json('/dev/stdin') SELECT status") != "Accepted" ]] ; then 
```

Comment:
note how I use the just-built binary here to parse the JSON reponse from `xcrun notarytool` :)